subsys/bluetooth/host/l2cap.c

```diff
@@ -1942,74 +2159,62 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 	/* Save state so it can be restored if we failed to send */
 	net_buf_simple_save(&buf->b, &state);
 
-	if ((buf->len <= ch->tx.mps) &&
-	    (net_buf_headroom(buf) >= BT_L2CAP_BUF_SIZE(0))) {
-		LOG_DBG("len <= MPS, not allocating seg for %p", buf);
-		seg = net_buf_ref(buf);
-
-		len = seg->len;
-	} else {
-		LOG_DBG("allocating segment for %p (%u bytes left)", buf, buf->len);
-		seg = l2cap_alloc_seg(ch);
-		if (!seg) {
-			LOG_DBG("failed to allocate seg for %p", buf);
-			atomic_inc(&ch->tx.credits);
-
-			return -EAGAIN;
-		}
-
-		/* Don't send more than TX MPS */
-		len = MIN(net_buf_tailroom(seg), ch->tx.mps);
-
-		/* Limit if original buffer is smaller than the segment */
-		len = MIN(buf->len, len);
+	seg = get_seg(buf, ch->tx.mps, ch);
 
-		net_buf_add_mem(seg, buf->data, len);
-		net_buf_pull(buf, len);
+	if (!seg) {
+		LOG_ERR("The impossible happened..");
```

Comment:
If this is impossible, why not an assert?

Reply:
https://github.com/jori-nordic/zephyr/blob/a6e401b2f2a07e5c08efc48e46284e5ddbc18954/subsys/bluetooth/host/conn.c#L704

Without joking tho, this is just a holdover of me trying to exercise this code branch. converted to `CHECKIF`.

Reply:
:D 

As you've previously mentioned, there are some clang/coverity issues with __ASSERT as well